waku/factory/node_factory.nim

```diff
@@ -372,4 +372,56 @@ proc startNode*(node: WakuNode, conf: WakuNodeConf,
   if conf.relay:
     node.peerManager.start()
 
-  return ok()
\ No newline at end of file
+  return ok()
+
+proc setupNode*(conf: WakuNodeConf, rng: Option[ref HmacDrbgContext] = none(ref HmacDrbgContext)):
+  Result[WakuNode, string] =
+    var nodeRng = if rng.isSome(): rng.get() else: crypto.newRng()
+
+    # Use provided key only if corresponding rng is also provided
+    let key =
+      if conf.nodeKey.isSome() and rng.isSome():
+        conf.nodeKey.get()
+      else:
+          warn "missing key or rng, generating new set"
+          crypto.PrivateKey.random(Secp256k1, nodeRng[]).valueOr:
+            error "Failed to generate key", error=error
+            return err("Failed to generate key: " & $error)
+      
+    let netConfig = networkConfiguration(conf, clientId).valueOr:
+      error "failed to create internal config", error=error
```

Comment:
Would it be useful to create an Error enum for all the possible errors?

For e.g. https://github.com/waku-org/nwaku/blob/1274b15d75d4150889623a0b52dfcb7afd1d48ae/waku/waku_archive/common.nim#L69-L75

Reply:
If you impl. `$` on the error enum then you could do
```
let myvar = doSomething().valueOr:
   error "this failed bla bla", error = $error
   retrun err($error)
```

IDK maybe not worth it.

Reply:
Yeah I don't think it's worth it for this specific case - It's nicer but I prefer keeping it as simple as possible :)